File: dashboard/server/src/data/model/raw/problem.rs
///! Definitions of problem model structures

use serde::Serialize;

#[derive(Serialize, Debug, Clone)]
pub struct InstanceInfo {
    pub id: String,

    #[serde(rename = "ref")]
    pub reference: String,
    pub jobs: usize,
    pub machines: usize,
    pub lower_bound: usize,
    pub lower_bound_ref: String,
    pub best_solution: usize,
    pub best_solution_ref: String,
    pub best_solution_time: String,
    pub best_solution_time_ref: String,
}

File: dashboard/client/src/model/problem.ts
export type InstanceInfo = {
  id: string,
  ref: string,
  jobs: number,
  machines: number,
  lower_bound: number,
  lower_bound_ref: string,
  best_solution: number,
  best_solution_ref: string,
  best_solution_time: string,
  best_solution_time_ref: string,
};

export type ExperimentConfig = {
  inputFile: string,
  outputDir: string,
  configFile?: string,
  nSeries: number,
};

export type Experiment = {
  name: string,
  instance: InstanceInfo,
  config: ExperimentConfig,
};

export type SolverConfig = {
  inputFile?: string,
  outputDir?: string,
  nGen: number,
  popSize?: number,
  delayConstFactor?: number,
  solverType?: string,
};

export type BatchConfig = {
  outputDir?: string,
  configs: Experiment[],
  solverConfig?: SolverConfig,
  startTime?: string,
};
